tests/MongoDB.Driver.Core.Tests/Core/NativeLibraryLoader/NativeLibraryLoaderTests.cs

```diff
@@ -26,10 +26,14 @@ namespace MongoDB.Driver.Core.Tests.Core.NativeLibraryLoader
     public class NativeLibraryLoaderTests
     {
         [Theory]
-        [InlineData(null, "mongo-csharp-driver")] // the default assembly-based logic
         [InlineData("mongo-csharp-driver", "mongo-csharp-driver")]
         [InlineData("mongo csharp driver", "mongo csharp driver")]
         [InlineData("&mongo$csharp@driver%", "&mongo$csharp@driver%")]
+
+        // the default assembly-based logic. Ideally expectedRootTestFolder should be mongo-csharp-driver,
+        // but since it's not mocked logic, it limits us where we can run our tests from. Avoid it by
+        // making a test assertation less strict
+        [InlineData(null, "")]
```

Comment:
Consider using the suggested refactoring: https://github.com/MikalaiMazurenka/mongo-csharp-driver/commit/bd94d56be40f2e154c48f85b33795cc68592b7cf

Reply:
the validation and asserting steps here are the same, the only difference is input data, the ideal solution here is `ClassData` but it looks like overkill for this case, a simple `if` solves everything we need to. So I would leave this as is

Reply:
Alright, then could you remove the extra blank line and consider changing the 3 line comment to
```suggestion
        [InlineData(null, "")] // This case validates only the ending of the path with GetBaseAssemblyUri being used
```

Reply:
I wanted to explain why this change is necessary since it's kind of not obvious exception, you comment doesn't provide this info, however I don't mind about changing messages, it will be good to hear @JamesKovacs  opinion here also